src/volatile_memory.rs

```diff
@@ -103,6 +108,33 @@ pub fn compute_offset(base: usize, offset: usize) -> Result<usize> {
     }
 }
 
+/// Trait for objects for which a reference can be extracted safely out of a
+/// VolatileSlice.  Objects that implement this trait must consist exclusively
+/// of atomic types from std::sync::atomic, except for AtomicPtr<T>.
+pub unsafe trait AtomicValued: Sync + Send {}
```

Comment:
I imagine `AtomicValued` mainly functions as a marker trait, but would it be useful to have `AtomicValued: ByteValued` instead of only `Sync + Send`? I guess that way we can use all those functions with `T: ByteValued` trait bounds.

Reply:
Unfortuanately that's not possible because `std::sync::atomic::Atomic*` types are not `Copy`, which `ByteValued` requires.

Reply:
But I think it's better this way, as it enforces that `AtomicValued` types are accessed with the proper memory orderings, and not via `VolatileRef` and friends.